hammocking/hammocking.py

```diff
@@ -149,7 +149,9 @@ def parse(self, input: Union[Path, str]) -> None:
         self.logger.debug(f"Command arguments: {parseOpts['args']}")
         for child in translation_unit.cursor.get_children():
             if child.spelling in self.symbols:
-                self.writer.add_header(str(child.location.file))
+                in_header = child.location.file.name != translation_unit.spelling
+                if in_header:  # We found it in the Source itself. Better not include the whole source!
```

Comment:
Avoid including the whole translation unit (aka C file).
This happens when someone borrows symbols by putting `"extern <variable>"` into his C file.